packages/app/src/background/services/connection/ConnectionService.ts

```diff
@@ -110,7 +151,50 @@ export default class ConnectionService extends BaseService implements IBackupabl
     return payload;
   };
 
-  getConnections = (): Map<string, IIdenityConnection> => this.connections;
+  getConnections = (): Record<string, IIdentityConnection> => Object.fromEntries(this.connections.entries());
+
+  connectRequest = async (_: unknown, { urlOrigin }: IZkMetadata): Promise<void> => {
+    await this.browserController.openPopup({ params: { redirect: Paths.CONNECT_IDENTITY, urlOrigin } });
+  };
+
+  getConnectedIdentity = (urlOrigin: string): ZkIdentitySemaphore | undefined => {
+    const connection = this.connections.get(urlOrigin);
+
+    if (!connection) {
+      return undefined;
+    }
+
+    return this.zkIdenityService.getIdentity(connection.commitment);
+  };
+
+  revealConnectedIdentityCommitmentRequest = async (_: unknown, { urlOrigin }: IZkMetadata): Promise<void> => {
```

Comment:
It's not correct to use connection service inside the identity service. The logic is the same as injector service currently has.

Reply:
Okay so the `ZkIdentity` would be only getting the identity regardless of any connections related info. 

But then can we only have `getConnectedIdentity` from the connection service and use it inside the Group service. 
```ts
// Connection Service
getConnectedIdentity = (urlOrigin: string): IIdentityData | undefined => {
    const connection = this.connections.get(urlOrigin);

    if (!connection) {
      return undefined;
    }

    const identity = this.zkIdenityService.getIdentity(connection.commitment);
    const commitment =  bigintToHex(identity.genIdentityCommitment());
    const metadata = connectedIdenity.metadata
    // The return could be more generic so that I can fit all the purposes for the other services like in Groups Service
    return {
      commitment ,
      metadata,
    };
  };
```


```ts
// Group Service

checkGroupMembership = async (
    { groupId }: ICheckGroupMembershipArgs,
    { urlOrigin }: IZkMetadata,
  ): Promise<boolean> => {
    const connectedIdenity = this.connectionService.getConnectedIdentity(urlOrigin!);
    
    if (!connectedIdenity) {
      throw new Error("No connected identity found");
    }

    return this.bandadaSevice.checkGroupMembership({ groupId, identity });
  };
```

Reply:
Group service uses `this.connectionService.getConnectedIdentity` inside private method which is reused across other methods. There is no need to return commitment and metadata from connection service only to cover group service case.